specification/apicenter/ApiCenter.Management/models.tsp

```diff
@@ -557,3 +589,88 @@ model DeletedServiceProperties {
   @doc("UTC date and time when the service was soft-deleted. The date conforms to the following format: yyyy-MM-ddTHH:mm:ssZ as specified by the ISO 8601 standard.")
   softDeletionDate?: utcDateTime;
 }
+
+#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-provisioning-state" "Existing service"
+@added(Versions.v2024_06_01_preview)
+@doc("API source properties.")
+model ApiSourceProperties {
+  @doc("Indicates if the specification should be imported along with metadata.")
+  shouldImportSpec?: ShouldImportSpec = ShouldImportSpec.OnDemand;
+
+  @doc("API source configuration for Azure API Management.")
+  apimSource?: ApimSource;
+}
+
+/**
+ * The API source specification import options.
+ */
+@added(Versions.v2024_06_01_preview)
+union ShouldImportSpec {
+  string,
+
+  /**
+   * Indicates that the specification should be never be imported.
+   */
+  Never: "never",
+
+  /**
+   * Indicates that the specification should be imported only by request.
+   */
+  OnDemand: "ondemand",
+
+  /**
+   * Indicates that the specification should always be imported along with metadata.
+   */
+  Always: "always",
+}
+
+/**
+ * API source configuration for Azure API Management.
+ */
+model ApimSource {
+  /**
+   * API Management service resource ID.
+   */
+  resourceId: string;
+
+  /**
+   * The resource ID of the managed identity that has access to the API Management instance.
+   */
+  msiResourceId?: string;
```

Comment:
>msiResourceId [](http://example.com/codeflow?start=2&length=13)

If this a fully qualified ARM Id ? if so, please use armResourceIdentifier instead of string

Reply:
Addressed.